com.unity.shadergraph/Editor/Data/Nodes/AbstractMaterialNode.cs

```diff
@@ -97,6 +97,10 @@ public virtual bool canSetPrecision
             get { return true; }
         }
 
+        // this is the precision after the inherit/automatic behavior has been calculated
+        // it does NOT include fallback to any graph default precision
+        public GraphPrecision graphPrecision { get; set; } = GraphPrecision.Single;
```

Comment:
Naming (both for the Enum and the field) could use a tweak here maybe as I definitely was assuming the latter before reading the comment, might be confusing for comprehension if scanning at a glance

Maybe 'InheritedPrecision' or 'DerivedPrecision' or something like that? 

Not a huge deal though as the comment clarifies it

Reply:
Hmm I see the problem as its also used at a SubGraph level, not sure what a more suitable name is thats generic across all these use cases

Reply:
Yeah..   I think technically to make it more clear we could separate this enum into two enums, to make clear the difference between pre-graph fallback and post-graph fallback.   We're currently using the GraphPrecision for both, as they are very similar:     pre-fallback the enum lets the node choose between:  force single, force half, or use the graph default precision.

post-fallback application, the enum effectively captures the subgraph view of precision:  force single, force half, or switchable (use $precision)
for shadergraphs, switchable is not allowed -- switchable is assumed to be single.

Reply:
Think it might be overkill for now though, we can always change the names later on.   :)